crates/ruff_linter/src/rules/ruff/rules/invalid_formatter_suppression_comment.rs

```diff
@@ -0,0 +1,278 @@
+///
+/// ## Examples
+/// In the following example, all suppression comments would cause
+/// a rule violation.
+///
+/// ```python
+/// def decorator():
+///     pass
+///
+///
+/// @decorator
+/// # fmt: off
+/// def example():
+///     if True:
+///         # fmt: skip
+///         expression = [
+///             # fmt: off
+///             1,
+///             2,
+///         ]
+///         # yapf: disable
+///     # fmt: on
+///     # yapf: enable
+/// ```
+#[violation]
+pub struct InvalidFormatterSuppressionComment {
+    reason: IgnoredReason,
+}
+
+impl AlwaysFixableViolation for InvalidFormatterSuppressionComment {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "This suppression comment is invalid because {}",
+            self.reason
+        )
+    }
+
+    fn fix_title(&self) -> String {
+        format!("Remove this comment")
+    }
+}
+
+/// RUF028
+pub(crate) fn ignored_formatter_suppression_comment(checker: &mut Checker, suite: &ast::Suite) {
+    let indexer = checker.indexer();
+    let locator = checker.locator();
+    let comment_ranges: SmallVec<[SuppressionComment; 8]> = indexer
+        .comment_ranges()
+        .into_iter()
+        .filter_map(|range| {
+            Some(SuppressionComment {
+                range: *range,
+                kind: SuppressionKind::from_comment(locator.slice(range))?,
+            })
+        })
+        .collect();
+
+    if comment_ranges.is_empty() {
+        return;
+    }
+
+    let mut comments = UselessSuppressionComments::new(locator);
+
+    let visitor = SuppressionCommentVisitor::new(
+        comment_ranges.into_iter(),
+        &mut comments,
+        checker.locator(),
+    );
+
+    visitor.visit(suite);
+
+    comments.sort();
+
+    for (range, reason) in comments.ignored_comments() {
+        checker.diagnostics.push(
+            Diagnostic::new(InvalidFormatterSuppressionComment { reason }, range)
+                .with_fix(Fix::unsafe_edit(delete_comment(range, checker.locator()))),
+        );
+    }
+}
+
+struct UselessSuppressionComments<'src, 'loc> {
+    captured: Vec<(TextRange, IgnoredReason)>,
+    locator: &'loc Locator<'src>,
+}
+
+impl<'src, 'loc> UselessSuppressionComments<'src, 'loc> {
+    fn new(locator: &'loc Locator<'src>) -> Self {
+        Self {
+            captured: vec![],
+            locator,
+        }
+    }
+    /// This function determines whether or not `comment` is a useful suppression comment.
+    /// If it isn't, it will give a reason why the comment is ignored. See [`IgnoredReason`] for more.
+    fn check_suppression_comment(
+        &self,
+        comment: &SuppressionCommentData,
+    ) -> Result<(), IgnoredReason> {
+        // check if the comment is inside of an expression.
+        if comment
+            .enclosing
+            .map(AnyNodeRef::is_expression)
+            .unwrap_or_default()
+        {
+            return Err(IgnoredReason::InsideExpression);
+        }
+
```

Comment:
We may need to change the condition to whether the enclosing statement is not a statement to also catch

```python
    def cls_method_a(
            # fmt: off
            cls) -> None:
        pass
```